### PR TITLE
New content for GPG encrypted Salt Pillars

### PR DESCRIPTION
**Pull request by 0rnela**
# Description

Action item to document encrypted Salt pillars was the outcome of research in https://github.com/SUSE/spacewalk/issues/14613.

# Target branches

Which documentation version does this PR apply to?

- [x] Master (Default)
- [x] Manager-4.2 https://github.com/uyuni-project/uyuni-docs/pull/1453
- [ ] Manager-4.1
- [ ] Manager-4.0

# Links

Fixes https://github.com/SUSE/spacewalk/issues/14853

### CODE REVIEW

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 1]
```diff
@@ -0,0 +1,113 @@
+[[salt-gpg-encrypted-pillars]]
+= GPG Encrypted Pillars
+
+{salt} has support for transparently decrypt GPG-encrypted Pillar data built-in.
```
Comment (jcayouette) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:4]: "has support to transparently decrypt"

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+The decryption happens on the {salt} Master.
+
+
+== Generate GPG keyring for {salt} Master
+
+The GPG keyring can be specified in [literal]```/etc/salt/master``` or in its own file under [literal]```/etc/salt/master.d/```, for example [literal]```/etc/salt/master.d/gpg-pillar.conf```. 
+
+Always create separate keyring for the {salt} Master.
```
Comment (jcayouette) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:12]: "Always create a separate keyring"

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+[[proc-key-pair-generation]]
+.Procedure: Generating key pair
+
+. On the {salt} Master create GPG home directory and restrict its permissions:
```
Comment (jcayouette) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:17]: create (the) GPG home directory

File: .changelog [hunk 4]
```diff
@@ -4,6 +4,7 @@
 # - Fixed error in Bat section of Upgrade Guide (bsc#1234567)
 # For guidelines: https://en.opensuse.org/openSUSE:Creating_a_changes_file_(RPM)#Changelog_section_.28.25changelog.29
 
+- Documented GPG encrypted Salt Pillars in the Specialized Guides, Salt book.
```
Comment (keichwa) [inline .changelog:7]:
```suggestion
- Documented GPG encrypted Salt pillars in the Specialized Guides, Salt book.
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 5]
```diff
@@ -0,0 +1,113 @@
+[[salt-gpg-encrypted-pillars]]
+= GPG Encrypted Pillars
+
+{salt} has support to transparently decrypt GPG-encrypted Pillar data built-in.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:4]:
```suggestion
{salt} has support to transparently decrypt GPG-encrypted pillar data built-in.
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+The decryption happens on the {salt} Master.
+
+
+== Generate GPG keyring for {salt} Master
+
+The GPG keyring can be specified in [literal]```/etc/salt/master``` or in its own file under [literal]```/etc/salt/master.d/```, for example [literal]```/etc/salt/master.d/gpg-pillar.conf```.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:10]:
```suggestion
The GPG keyring can be specified in [path]``/etc/salt/master`` or in its own file under [path]``/etc/salt/master.d/``, for example [path]``/etc/salt/master.d/gpg-pillar.conf``. 
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+----
+. Configure {salt} Master to use the new GPG home directory
++
+----
+echo 'gpg_keydir: /etc/salt/gpgkeys' >/etc/salt/master.d/gpg-pillar.conf
+systemctl reload-or-restart salt-master
+----
+
+
+== Use GPG for encrypting Pillar secrets
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:46]:
```suggestion
== Use GPG for encrypting pillar secrets
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+
+Salt GPG renderer decrypts GPG encrypted contents that are ASCI-armored. 
+To use the GPG renderer in a Pillar YAML file, change
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:49]:
```suggestion
To use the GPG renderer in a pillar YAML file, change 
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+
+Encrypting pillar secrets can be done anywhere as long as the GPG and the public key generated in <<proc-key-pair-generation>> are available. 
+
+In this example, "SUMA Salt Master" is the GPG key’s UID created earlier.
+----
+echo 't0ps3cr3t' | gpg --armor --batch --encrypt --recipient "SUMA Salt Master"
+----
+
+
+Once the GPG encrytped contents are created and available as ASCII-armored output, this output can be used as a multi-line string in a pillar yaml file like below.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:66]:
```suggestion
When the GPG encrytped contents are created and available as ASCII-armored output, this output can be used as a multi-line string in a pillar YAML file:
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 10]
```diff
@@ -0,0 +1,113 @@
...
+    GXU7OsiL1dO0/sST1Awo5i99kVt/kA6DCGDuxTNpLrauNLOKUbtwcxvavtNZGwdQ
+    yI9zWVx8qerWE0aO3M7zVDJftv77faV2ENiqzaadvtJHAZynW4GW7rSuP1RXFzlB
+    DOAmzdRuIJwiLC9R2BKu3x+avReQb6xoz7eF7WthC0H0dz4mYakwPlVZ5yqYa/+G
+    83i951rqAGI=
+    =g+ji
+    -----END PGP MESSAGE-----
+----
+
+
+Once the pillar is assigned to a system via the [literal]```top.sls```, the GPG encrypted pillar data is available in a decrypted format.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:87]:
```suggestion
When the pillar is assigned to a system with [path]``top.sls``, the GPG encrypted pillar data is available in a decrypted format. 
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+    -----END PGP MESSAGE-----
+----
+
+
+Once the pillar is assigned to a system via the [literal]```top.sls```, the GPG encrypted pillar data is available in a decrypted format. 
+
+
+[NOTE]
+====
+The minion’s in-memory cache is only updated on startup or when running execution module functions that trigger a cache refresh (for example [literal```saltutil.refresh_pillar```, [litaral]```pillar.items```, or [literal]```state.apply```)
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:92]:
```suggestion
The client's in-memory cache is only updated on startup or when running execution module functions that trigger a cache refresh such as [literal]``saltutil.refresh_pillar``, [litaral]``pillar.items``, or [literal]``state.apply``.
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+----
+
+
+== Export the GPG key
+
+To export the GPG key, use the command:
+----
+gpg --export 'SUMA Salt Master' --homedir /etc/salt/gpgkeys --output suma-salt-master.gpg
+----
+where 'SUMA Salt Master' is the name used during key generation.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:109]:
```suggestion
Here 'SUMA Salt Master' is the name used during key generation.
```

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+
+The [literal]```suma-salt-master.gpg``` public key can be freely shared.
+
+//For more information about GPG, see https://www.gnupg.org/documentation/.
```
Comment (keichwa) [inline modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:113]: Is this meant to be commented and thus invisible?

**Review by keichwa — CHANGES_REQUESTED** [review]
I hope I got it right.

File: modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc (hunk 13 above)
Comment (0rnela) [reply modules/specialized-guides/pages/salt/salt-gpg-encrypted-pillars.adoc:113]: yes

**Review by agraul — COMMENTED** [review]
Looks good to me. It is hard for me to gauge if anything is lacking for someone who doesn't not know GPG, maybe someone who considers themself new to GPG could chime in on that.

**Review by keichwa — APPROVED** [review]
Thank you!

**Pull request merged**